### PR TITLE
Remove problematic links

### PR DESCRIPTION
**Pull request by vhf**
@Devinsuit would you be kind enough to give this PR a quick review?

Thanks a lot!

### CODE REVIEW

File: free-programming-books-pl.md [hunk 1]
```diff
@@ -26,8 +26,6 @@
 * [Pisanie OS](https://pl.wikibooks.org/wiki/Pisanie_OS)
 * [Pro Git](http://git-scm.com/book/pl/)
 * [SVN](https://pl.wikibooks.org/wiki/Subversion)
-* [System kontroli wersji Subversion](http://svnbook.opensys.pl)
```
Comment (Devinsuit) [inline free-programming-books-pl.md]: http://www.gajdaw.pl/varia/subversion-system-kontroli-wersji-tutorial/

File: free-programming-books-pt_BR.md [hunk 2]
```diff
@@ -49,7 +49,6 @@
 ### C++
 
 * [Apostila Linguagem C++](http://www.ime.usp.br/~slago/slago-C++.pdf) - Silvio Lago (PDF)
-* [Programação Orientada a Objetos em C++](http://www.filoczar.com.br/sistemas_operacionais/linux/cpppara_linux.pdf) (PDF)
```
Comment (Devinsuit) [inline free-programming-books-pt_BR.md]: http://webserver2.tecgraf.puc-rio.br/~manuel/Download/Programacao%20Orientada%20a%20Objetos%20em%20C++.pdf

File: free-programming-books-ru.md [hunk 3]
```diff
@@ -251,7 +250,6 @@
 
 ### Rust
 
-* [Rust на примерах](https://gordon-f.github.io/rust-by-example-ru/)
```
Comment (Devinsuit) [inline free-programming-books-ru.md]: https://rurust.github.io/rust-by-example-ru/

File: free-programming-books-zh.md [hunk 4]
```diff
@@ -196,10 +195,7 @@
 * [命令行的艺术](https://github.com/jlevy/the-art-of-command-line/blob/master/README-zh.md)
 * [嵌入式 Linux 知识库 (eLinux.org 中文版)](https://tinylab.gitbooks.io/elinux/content/zh/)
 * [开源世界旅行手册](http://i.linuxtoy.org/docs/guide/index.html)
-* [深入分析Linux内核源码](http://www.kerneltravel.net/kernel-book/%E6%B7%B1%E5%85%A5%E5%88%86%E6%9E%90Linux%E5%86%85%E6%A0%B8%E6%BA%90%E7%A0%81.html)
 * [理解Linux进程](https://github.com/tobegit3hub/understand_linux_process)
-* [鸟哥的 Linux 私房菜 基础学习篇](http://vbird.dic.ksu.edu.tw/linux_basic/linux_basic.php)
```
Comment (Devinsuit) [inline free-programming-books-zh.md]: http://cn.linux.vbird.org/linux_basic/linux_basic.php

File: free-programming-books-zh.md [hunk 5]
```diff
@@ -196,10 +195,7 @@
...
-* [鸟哥的 Linux 私房菜 服务器架设篇](http://vbird.dic.ksu.edu.tw/linux_server/)
```
Comment (Devinsuit) [inline free-programming-books-zh.md]: http://cn.linux.vbird.org/linux_server/

File: free-programming-books-zh.md [hunk 6]
```diff
@@ -232,7 +228,6 @@
 * [Git教程](http://www.liaoxuefeng.com/wiki/0013739516305929606dd18361248578c67b8067c8c017b000) （本文由 [@廖雪峰](http://weibo.com/liaoxuefeng) 创作，如果觉得本教程对您有帮助，可以去 [iTunes](https://itunes.apple.com/cn/app/git-jiao-cheng/id876420437) 购买）
 * [Got GitHub](https://github.com/gotgit/gotgithub)
 * [GotGitHub](http://www.worldhello.net/gotgithub/index.html)
-* [HgInit (中文版)](http://bucunzai.net/hginit/)
```
Comment (Devinsuit) [inline free-programming-books-zh.md]: https://zh-hginit.readthedocs.io/en/latest/

File: free-programming-books-zh.md [hunk 7]
```diff
@@ -254,7 +249,6 @@
 * [ELKstack 中文指南](http://kibana.logstash.es)
 * [Logstash 最佳实践](https://github.com/chenryn/logstash-best-practice-cn)
 * [Mastering Elasticsearch(中文版)](http://udn.yyuap.com/doc/mastering-elasticsearch/)
-* [Puppet 2.7 Cookbook 中文版](http://bbs.konotes.org/workdoc/puppet-27/)
```
Comment (Devinsuit) [inline free-programming-books-zh.md]: https://www.gitbook.com/book/wizardforcel/puppet-27-cookbook/details

File: free-programming-books-zh.md [hunk 8]
```diff
@@ -316,7 +310,6 @@
 
 * [Android Design(中文版)](http://www.apkbus.com/design/index.html)
 * [Android Note(开发过程中积累的知识点)](https://github.com/CharonChui/AndroidNote)
-* [Android6.0新特性详解](http://leanote.com/blog/post/561658f938f41126b2000298?hmsr=toutiao.io&utm_medium=toutiao.io&utm_source=toutiao.io)
```
Comment (Devinsuit) [inline free-programming-books-zh.md]: It's working now.

File: free-programming-books-zh.md [hunk 9]
```diff
@@ -405,12 +398,10 @@
 
 ### Golang
 
-* [Effective Go](http://www.hellogcc.org/effective_go.html)
 * [Go Web 编程](https://github.com/astaxie/build-web-application-with-golang)
 * [Go 入门指南](https://github.com/Unknwon/the-way-to-go_ZH_CN) (《The Way to Go》中文版)
 * [Go 官方文档翻译](https://github.com/golang-china/golangdoc.translations)
-* [Go 指南](http://go-tour-zh.appsp0t.com) (《A Tour of Go》中文版)
```
Comment (Devinsuit) [inline free-programming-books-zh.md]: https://tour.go-zh.org/list

File: free-programming-books-zh.md [hunk 10]
```diff
@@ -578,15 +569,14 @@
...
-* [The NodeJS 中文文档](https://www.gitbook.com/book/0532/nodejs/details)（社区翻译）
+* [The NodeJS 中文文档](https://www.gitbook.com/book/0532/nodejs/details) （社区翻译）
 * [七天学会NodeJS](http://nqdeng.github.io/7-days-nodejs/) 阿里出品，很好的入门资料
 
 
 ### Perl
 
 * [Master Perl Today](https://github.com/fayland/chinese-perl-book)
 * [《Modern Perl》中文版](https://github.com/horus/modern_perl_book)
-* [Perl 5 教程](http://www.cbi.pku.edu.cn/chinese/documents/perl/index.htm)
```
Comment (Devinsuit) [inline free-programming-books-zh.md]: http://net.pku.edu.cn/~yhf/tutorial/perl/perl.html

File: problem-sets-competitive-programming.md [hunk 11]
```diff
@@ -43,7 +43,6 @@
 * [Anarchy Golf](http://golf.shinh.org)
 * [CareerCup](http://www.careercup.com)
 * [CheckIO](http://www.checkio.org)
-* [Codeforces](http://codeforces.com/problemset)
```
Comment (Devinsuit) [inline problem-sets-competitive-programming.md]: It's working now.

File: free-programming-books.md [hunk 12]
```diff
@@ -1163,7 +1149,6 @@
 * [Real World Haskell](http://book.realworldhaskell.org)
 * [Speeding Through Haskell](http://www.sthaskell.com)
 * [The Haskell Road to Logic, Math and Programming](https://fldit-www.cs.uni-dortmund.de/~peter/PS07/HR.pdf) (PDF)
-* [The Haskell School of Music](http://haskell.cs.yale.edu/?post_type=publication&p=112) (PDF) (work in progress)
```
Comment (vhf) [inline free-programming-books.md]: Copyright notice isn't that friendly, I'll remove this link.

**Pull request merged**